### PR TITLE
Re-land experiment runner tests

### PR DESCRIPTION
**Pull request by frgossen**
The previous attempt at landing these tests (e.g. #6093) relied on torchbench, which is not available in the CI setup. Instead, rely on the dummy benchmarks.

### CODE REVIEW

File: benchmarks/benchmark_experiment.py [hunk 1]
```diff
@@ -84,6 +84,7 @@ def is_available(self, experiment_config):
     if experiment_config["dynamo"] == "inductor" and not (
         experiment_config["accelerator"] == "cuda" and
         not experiment_config["xla"]):
+      logger.warning("No Inductor")
```
Comment (cota) [inline benchmarks/benchmark_experiment.py:87]: Please remove this if the PR ends up passing CI

File: benchmarks/benchmark_experiment.py [hunk 2]
```diff
@@ -98,6 +99,7 @@ def is_available(self, experiment_config):
       return False
     if (experiment_config["accelerator"] == "cuda" and
         not experiment_config["xla"] and not torch.cuda.is_available()):
+      logger.warning("No Inductor because cuda isn't available")
```
Comment (cota) [inline benchmarks/benchmark_experiment.py:102]: And this

**Review by cota — APPROVED** [review]

**Comment by vanbasten23** [comment]
Can you add some context on what experiment runner tests failure you are fixing?

File: benchmarks/benchmark_experiment.py (hunk 1 above)
Comment (cota) [reply benchmarks/benchmark_experiment.py]: Done

File: benchmarks/benchmark_experiment.py (hunk 2 above)
Comment (cota) [reply benchmarks/benchmark_experiment.py]: Done

**Comment by cota** [comment]
> Can you add some context on what experiment runner tests failure you are fixing?

Good point. I've edited the PR message at the top and also edited the PR title accordingly.

**Pull request merged**